std/os/index.zig

```diff
@@ -532,6 +536,8 @@ pub fn posixDup2(old_fd: i32, new_fd: i32) !void {
                 posix.EBUSY, posix.EINTR => continue,
                 posix.EMFILE => error.ProcessFdQuotaExceeded,
                 posix.EINVAL => unreachable,
+                posix.EIO => unreachable,
```

Comment:
why is this one unreachable? is it documented to happen only when you give bad inputs to dup2?

Reply:
This is an error on my part. EIO is documented in case of filesystem errors, although I am not sure in which conditions it can be triggered.

Reply:
In this case I think it should be exposed as a possible error. If dup2 can cause a file system error, then zig code should be able to handle that error.